src/test/kotlin/uk/gov/dluhc/printapi/rest/aed/UpdateAnonymousElectorDocumentByApplicationIdIntegrationTest.kt

```diff
@@ -92,4 +102,151 @@ internal class UpdateAnonymousElectorDocumentByApplicationIdIntegrationTest : In
             .hasError("Not Found")
             .hasMessage("Certificate for eroId = $ERO_ID with sourceType = ANONYMOUS_ELECTOR_DOCUMENT and sourceReference = $unknownSourceReference not found")
     }
+
+    @Test
+    fun `should update elector's email and phone number`() {
+        // Given
+        val eroResponse = buildElectoralRegistrationOfficeResponse(
+            id = ERO_ID,
+            localAuthorities = listOf(buildLocalAuthorityResponse(gssCode = GSS_CODE), buildLocalAuthorityResponse())
+        )
+        wireMockService.stubCognitoJwtIssuerResponse()
+        wireMockService.stubEroManagementGetEroByEroId(eroResponse, ERO_ID)
+        val originalEmailAddress = aValidEmailAddress()
+        val originalPhoneNumber = aValidPhoneNumber()
+        val aed = buildAnonymousElectorDocument(
+            gssCode = GSS_CODE,
+            sourceReference = APPLICATION_ID,
+            contactDetails = buildAedContactDetails(email = originalEmailAddress, phoneNumber = originalPhoneNumber)
+        )
+        anonymousElectorDocumentRepository.save(aed)
+        Thread.sleep(2000)
+
+        val dateCreated = aed.contactDetails!!.dateCreated!!
+        val createdBy = aed.contactDetails!!.createdBy!!
+        val dateUpdated = aed.contactDetails!!.dateUpdated!!
+        val expectedUpdatedBy = "an-ero-user1@$ERO_ID.gov.uk"
+        val newEmailAddress = anotherValidEmailAddress()
+        val newPhoneNumber = anotherValidPhoneNumber()
+        val updateAedRequest = buildUpdateAnonymousElectorDocumentRequest(
+            sourceReference = APPLICATION_ID,
+            email = newEmailAddress,
+            phoneNumber = newPhoneNumber
+        )
+
+        // When
+        webTestClient.patch()
+            .uri(URI_TEMPLATE, ERO_ID, APPLICATION_ID)
+            .bearerToken(getVCAnonymousAdminBearerToken(eroId = ERO_ID))
+            .contentType(APPLICATION_JSON)
+            .withBody(updateAedRequest)
+            .exchange()
+            .expectStatus().isNoContent
+
+        // Then
+        val updated = anonymousElectorDocumentRepository.findById(aed.id!!).get()
+        assertThat(updated.contactDetails!!.email).isEqualTo(newEmailAddress)
```

Comment:
Could use `AnonymousElectorDocumentCertificateAssert` to assert these values? (same comment for test below)

Reply:
Happy to do that for email and phone number 👍  

I could do it for most of the others, though the test for `dateUpdated` is different in each case.